### PR TITLE
Replace java based self-signed certificate generation with OpenSSL

### PR DESCRIPTION
**Pull request by alexnb**
Replace java based self-signed certificate generation with instructions on how to use OpenSSL.

Certificate generation using the `agent` binary will remain since it does not have security issues.

----

If you open a PR that needs to go into a current version, you need to *cherry-pick your commit from dev over to the current version branch*. Only then will the proper builds that generate html/pdf be run. But beware: Docs will be generated but not published automatically!

- [X] N/A - or - I have added the appropriate "cherry-pick-to" labels to this PR so I don't forget to do this later!

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Comment by alexnb** [comment]
This is the first part of my documentation changes. Changes related to "Docker first look" will be in a separate PR.

File: modules/ROOT/pages/addition/agent-installation/agent-self-signed-certificates.adoc [hunk 1]
```diff
@@ -44,7 +44,7 @@ options (required options marked with *):
 
 If you are generating a certificate for an agent on `localhost`, `localhost` is the primary name set as Subject Name field on the certificate and is also used as the name of generated files. If `CONFIG_AGENT_NAME` is not specified, the Subject Name in agent client certificate along with the host information of agent will be used as the agent name.
 
-In order to generate a self-signed certificate for the above example, execute the following command;
+In order to generate a self-signed certificate for the above example, execute the following command:
```
Comment (SaschaPeukert) [inline modules/ROOT/pages/addition/agent-installation/agent-self-signed-certificates.adoc:47]: Should this file not be removed or changed to: We don't do ssc anymore please see. other page.. ?

**Review by SaschaPeukert — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/addition/agent-installation/agent-self-signed-certificates.adoc (hunk 1 above)
Comment (alexnb) [reply modules/ROOT/pages/addition/agent-installation/agent-self-signed-certificates.adoc:47]: Our `ssc` tool in the _agent_ is still available. It doesn't use weak algorithms. As I spoke with Sebastian, we decided to keep it.

It is also used in our build process and I didn't want to invest too much time to replace it with `openssl` there too.

OK or do you think we should wipe it too?

Comment (SaschaPeukert) [reply modules/ROOT/pages/addition/agent-installation/agent-self-signed-certificates.adoc:47]: Okay, thanks for pointing that out. I could see how that might be confusing to our users but for now I am happy with you keeping it then

**Review by SaschaPeukert — APPROVED** [review]

File: modules/ROOT/pages/first-look/docker-first-look.adoc [hunk 2]
```diff
@@ -23,21 +23,18 @@ You need to set up the following host names in your `/etc/hosts` or `C:/windows/
...
+=== Generate self-signed certificates
 
+. Create a directory *`~/.nom/ssc`* on your local machine before running Docker compose.
+The compose files below specify that this directory will be mounted into the docker container to make the generated certificates available to the NOM server and the agent.
++
 [NOTE]
 ====
 For *`sudo`* docker compose command, make sure the correct path to *`.nom/ssc`* is set in the bind section below.
 ====
+. Change to *`~/.nom/ssc`* and generate self-signed certificates as described *xref:installation/self-signed-certificate.adoc[here]*, using `server` as common name (CN) and DNS name and `127.0.0.1` as IP address.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/first-look/docker-first-look.adoc]:
```suggestion
. Change to *`~/.nom/ssc`* and generate self-signed certificates as described *xref:installation/self-signed-certificate.adoc[here]*, using `server` as common name (CN) and DNS name, and `127.0.0.1` as IP address.
```

File: modules/ROOT/pages/first-look/docker-first-look.adoc [hunk 3]
```diff
@@ -23,21 +23,18 @@ You need to set up the following host names in your `/etc/hosts` or `C:/windows/
...
+This will result in `server.cer` and `server.pfx` being created.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/first-look/docker-first-look.adoc]:
```suggestion
This results in the creation of `server.cer` and `server.pfx`.
```

File: modules/ROOT/pages/installation/self-signed-certificate.adoc [hunk 4]
```diff
@@ -8,64 +8,54 @@ For production environments, it is advisable to use a trusted certificate issuer
 This section outlines a practical way to generate a self-signed certificate for test and demo purposes.
 ====
 
-The NOM Server contains a utility to generate a self-signed certificate suitable for your environment.
-
-To invoke the utility and get help on usage, navigate to the NOM Server folder and execute the following command:
-
+The following instructions show how a self-signed certificate suitable for a NOM environment can be generated using the OpenSSL library.
+Compatible self-signed certificates, generated using other libraries or online tools will also work with NOM.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/installation/self-signed-certificate.adoc]:
```suggestion
Compatible self-signed certificates, generated using other libraries or online tools also work with NOM.
```

**Review by AlexicaWright — COMMENTED** [review]
Some editorial comments.

**Pull request merged**